lib/subjects/selector.rb

```diff
@@ -104,15 +107,19 @@ def subjects_page_size
     end
 
     def context
-      @context ||= {
-        workflow: workflow,
-        user_seen: UserSeenSubject.where(user: user, workflow: workflow),
-        finished_workflow: user&.has_finished?(workflow)
-      }
+      @context ||= { workflow: workflow, user: user }
     end
 
     def subject_set_id
       params[:subject_set_id]
     end
+
+    def subject_selection_context
+      if Panoptes.flipper[:skip_subject_selection_context].enabled?
```

Comment:
This will make a db call per selection request, we can look at adding a redis cache layer in to Flipper as well if this is too costly.